source/MaterialXTest/MaterialXGenOsl/GenOsl.cpp

```diff
@@ -102,6 +104,88 @@ TEST_CASE("GenShader: OSL Unique Names", "[genosl]")
     GenShaderUtil::testUniqueNames(context, mx::Stage::PIXEL);
 }
 
+TEST_CASE("GenShader: Metadata", "[genosl]")
+{
+    mx::FilePath searchPath = mx::FilePath::getCurrentPath() / mx::FilePath("libraries");
+
+    mx::DocumentPtr doc = mx::createDocument();
+    mx::loadLibraries({ "stdlib", "pbrlib", "bxdf" }, searchPath, doc);
+
+    //
+    // Define custom attributes to be exported as shader metadata
+    //
+
+    mx::AttributeDefPtr adNodeName = doc->addAttributeDef("AD_node_name");
```

Comment:
If you have attribute defs at the document level, when you import a document will the data be properly imported? Not sure.

Reply:
Is not all element types imported by default when you import a document into another? AttributeDefs can only exist at the root level.. I'm not sure what happens if there is a name conflict, but it should not be different from any other element type.